invenio_communities/utils.py

```diff
@@ -109,3 +112,15 @@ def on_datastore_post_commit(sender, session):
             users = role.users.all()
             for user in users:
                 on_user_membership_change(Identity(user.id))
+
+
+def humanize_byte_size(size):
+    """Converts bytes to largest unit (e.g., KB, MB, GB)."""
+    s = Decimal(size)
+    for unit in ["B", "KB", "MB", "GB", "TB", "PB"]:
+        if s < BYTES_PER_UNIT:
+            with localcontext() as ctx:
+                ctx.rounding = ROUND_HALF_UP
+                s = s.quantize(Decimal("0.00"))
+            return s, unit
+        s /= BYTES_PER_UNIT
```

Comment:
A couple last things:

- If we are going the `quantize` route, we may as well pass the rounding to simplify things: `s = s.quantize(q, rounding=ROUND_HALF_UP)` (`q = Decimal("0.00")` once and outside the loop) and remove all the context stuff
- Let's keep `BYTES_PER_UNIT` in the function. Was there a particular reason why out? (keeping it out exposes us to problems that we shouldn't have to care about)

Reply:
- Good call on the first point
- Honestly, the only reason I put it outside of the function was that I thought it was the standard way to place constants. I'm fine with scoping it to a function. For my own understanding, are we doing this to avoid naming collisions or were you thinking of something else?